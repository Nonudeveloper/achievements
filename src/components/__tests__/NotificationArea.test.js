import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import NotificationArea from "../NotificationArea";

describe("<NotificationArea>", () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("Should test Notification component", () => {
    const component = shallow(
      <NotificationArea handleClose={() => {}} message="test" open={true} />
    );

    expect(component).toMatchSnapshot();
  });

  it("Should render Notification message", () => {
    const component = shallow(
      <NotificationArea handleClose={() => {}} message="test" open={true} />
    );

    expect(component.prop("message")).toEqual(<span>test</span>);
  });
});
