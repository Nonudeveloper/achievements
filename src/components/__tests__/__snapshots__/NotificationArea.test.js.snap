// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationArea> Should test Notification component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationArea
    handleClose={[Function]}
    message="test"
    open={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "ContentProps": Object {
        "aria-describedby": "message-id",
      },
      "action": Array [
        <WithStyles(IconButton)
          aria-label="Close"
          color="inherit"
          onClick={[Function]}
>
          <pure(Close) />
</WithStyles(IconButton)>,
      ],
      "anchorOrigin": Object {
        "horizontal": "left",
        "vertical": "bottom",
      },
      "message": <span>
        test
</span>,
      "open": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ContentProps": Object {
          "aria-describedby": "message-id",
        },
        "action": Array [
          <WithStyles(IconButton)
            aria-label="Close"
            color="inherit"
            onClick={[Function]}
>
            <pure(Close) />
</WithStyles(IconButton)>,
        ],
        "anchorOrigin": Object {
          "horizontal": "left",
          "vertical": "bottom",
        },
        "message": <span>
          test
</span>,
        "open": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
